src/relayFeeCalculator/chain-queries/baseQuery.ts

```diff
@@ -213,7 +213,9 @@ export class QueryBase implements QueryInterface {
         ? Promise.resolve({ maxFeePerGas: _gasPrice })
         : getGasPriceEstimate(provider, { chainId, baseFeeMultiplier, priorityFeeMultiplier, transport, unsignedTx }),
     ] as const;
-    const [nativeGasCost, { maxFeePerGas: gasPrice }] = await Promise.all(queries);
+    const [nativeGasCost, _gasPriceEstimate] = await Promise.all(queries);
+    // It should be safe to cast to an EvmGasPriceEstimate here since QueryBase is only used for EVM chains.
```

Comment:
It's probably a good use case for an `assert(EvmGasPriceEstimate.is(gasPriceEstimate))`. We'd need superstruct defs for this.

Reply:
I agree. Probably another place for superstruct.